File: src/main/scala/resonant/lib/network/IPlayerUsing.scala
package resonant.lib.network

import java.util.Set

import net.minecraft.entity.player.EntityPlayer

/**
 * TPlayerUsing is a trait applied to all tiles that can have GUI. It is used to indicate which players the tile needs to send a packet too. The set provided will be mutated by the ContainerBase class in the GUI.
 */
trait IPlayerUsing
{
  def getPlayersUsing: Set[EntityPlayer]
}

File: src/main/scala/com/builtbroken/mc/api/tile/IPlayerUsing.java
package com.builtbroken.mc.api.tile;

import net.minecraft.entity.player.EntityPlayer;

import java.util.Collection;

/**
 * Created by robert on 1/12/2015.
 */
public interface IPlayerUsing
{
    Collection<EntityPlayer> getPlayersUsing();
}
